docs/getting-started-tutorial/7-using-actors-for-chat.adoc

```diff
@@ -0,0 +1,159 @@
+:idprefix:
+:idseparator: -
+:toc: right
+:toclevels: 2
+
+# Using Actors for Chats
+
+Actors are fairly simple: they receive messages, do something with them, and
+potentially respond to them. This isn't unlike the process that you go through
+when you call a method on a regular class; however, when you send a message to
+an actor, it is processed after any earlier messages you sent, never at the
+same time. This makes it a lot easier to reason about what's going on at any
+given point, even if multiple threads are sending messages to the actor.
+
+## Storing the Message List
+
+We're going to use a fairly basic actor for now. As before, we're going to have
+a single chat room for the entire site we're building, so we'll use a singleton
+object. Lift provides a very simple actor implementation, which we'll be using
+here. There are more complicated actor systems, like the one provided by
+http://aka.io[Akka], but they're only necessary in cases where you need more
+flexibility or fault tolerance. We'll stick to the easy stuff, starting with a
+new file at `src/main/scala/code/actor/ChatActor.scala`:
+
+```
+package code
+package actor
+
+import net.liftweb.actor._
+
+case class ChatMessage(poster: String, body: String)
+
+case class MessagePosted(message: ChatMessage)
+
+object ChatActor extends LiftActor {
+  private var messageEntries = List[ChatMessage]()
+
+  def messageHandler = {
+    case MessagePosted(newMessage) =>
+      messageEntries ::= newMessage
+  }
+}
+```
+
+This provides a very basic actor that can receive a new message and add it to
+its internal list. We've moved the `ChatMessage` class from the `Chat` snippet
+to this file. Typically, messages to actors are case classes. This is because
+they're easy to pattern match (as you can see, message handling is done via
+pattern matching footnote:[Strictly speaking, `messageHandler` is a
+`PartialFunction`. This means that it can match any subset of objects that it
+wants to.]) and because they're generally immutable, so there's no chance of
```

Comment:
Technically, there's `little` change of someone else trying to modify the message as we're processing it. But, eh, technicalities. heh.